### ISSUE TITLE
Can manifold-exceptions be used without manifold-rt in the runtime classpath?

### ISSUE THREAD

**Issue by rjb3977** [manifold-systems/manifold#193]
I tried including manifold-exceptions in a Gradle project using (pieces of) the snippet provided. Building it works just fine, and checked exceptions do indeed act like unchecked ones. However, trying to run the application (a simple test constructing and printing a URL) results in the following error:

```
Exception in thread "main" java.lang.NoClassDefFoundError: manifold/rt/api/IBootstrap
        at foo.Foo.<clinit>(Foo.java:6)
Caused by: java.lang.ClassNotFoundException: manifold.rt.api.IBootstrap
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 1 more
```

If I disable the Manifold compiler plugin (comment out the line with `-Xplugin:Manifold` in the build.gradle), this error doesn't occur, and checked exceptions behave as normal. That's to be expected, though.

If I add manifold-rt as an implementation dependency, everything works as expected. However, manifold-exceptions doesn't seem to say it requires a runtime dependency anywhere. Or do all manifold compile-time libraries require manifold-rt at runtime?

Attached is a minimal Gradle project demonstrating the issue. `gradlew build` works just fine, but `gradlew run` causes the error for me. Note that I'm using OpenJDK 14.0.1 from the AdoptOpenJDK project.

[foo.zip](https://github.com/manifold-systems/manifold/files/4833331/foo.zip)

**Comment by rsmckinney** [comment]
@rjb3977 Hi. For now, manifold-exceptions requires manifold-rt in the runtime classpath. *Should* it require manifold-rt?  No, there's really no need for it because there are no runtime calls corresponding with the use of manifold-exceptions... except the bootstrap call that is added to the class initializer of classes processed with manifold, which is unnecessary in this case, but is happening due to an unwanted transitive dependency on manifold-rt through manifold-exceptions. This happened recently with the changes to enable manifold on other platforms such as Android and Kotlin.  My bad for not catching this sooner.  Thanks for reporting it!  I should have it fixed in the next release.  For the interim (maybe a week) you'll need to include the manifold-rt implementation dependency (it's only 49k and has a near zero runtime impact).

**Comment by rsmckinney** [comment]
Fixed in release 2020.1.16

**Comment by rjb3977** [comment]
The issue doesn't appear to be fixed in 2020.1.16. It seems manifold-exceptions has a dependency on manifold-ext, which is still adding the bootstrap. Adding no-bootstrap does work though.

**Comment by rsmckinney** [comment]
Sorry about that.  Here's the deal.  That change and https://github.com/manifold-systems/manifold/commit/24d4581a61edafb986df68d167a76c2653779022 address an issue where a project **not** using `manifold-ext` does not include the runtime bootstrap in the class initialization block.  However, the `manifold-exception` straddles both worlds -- it doesn't reference any manifold runtime symbols, yet it requires classes to be processed with `manifold-ext`.  For this particular case you need to use the `no-bootstrap` plugin argument as I mentioned earlier.  That or include the `manifold-rt` dependency.  Personally, I prefer the former.

Although it didn't quite "fix" the issue as you wanted, your reporting it did result in purely static features such as manifold-properties, manifold-image, and manifold-strings, getting fixed to be independent of manifold-rt. So thanks again for reporting this!
